claat/cmd/util.go

```diff
@@ -43,36 +42,25 @@ const (
 	reportOk  = "ok\t%s"
 )
 
-var (
-	Exit      int               // program exit code
-	exitMu    sync.Mutex        // guards exit
-	ExtraVars map[string]string // Extra template variables passed on the command line.
-)
+var ExtraVars map[string]string // Extra template variables passed on the command line.
 
 // isStdout reports whether filename is stdout.
 func isStdout(filename string) bool {
 	return filename == stdout
 }
 
-// errorf calls log.Printf with fmt and args, and sets non-zero exit code.
-func errorf(format string, args ...interface{}) {
-	log.Printf(format, args...)
-	exitMu.Lock()
-	Exit = 1
-	exitMu.Unlock()
-}
-
 // ParseExtraVars parses extra template variables from command line.
 // extra is any additional arguments to pass to format templates. Should be formatted as JSON objects of string:string KV pairs.
-func ParseExtraVars(extra string) map[string]string {
+func ParseExtraVars(extra string) (map[string]string, error) {
 	vars := make(map[string]string)
 	if extra == "" {
-		return vars
+		return vars, nil
 	}
 	b := []byte(extra)
 	err := json.Unmarshal(b, &vars)
 	if err != nil {
-		errorf("Error parsing additional template data: %v", err)
+		log.Printf("Error parsing additional template data: %v", err)
```

Comment:
This pattern seems off to me. Now that we are returning an error, this function probably shouldn't print, especially since it's not a CLI function. I would recommend this:

```go
if err != nil {
  return nil, errors.Wrap(err, "failed to parsing additional template data")
}
```

and then later in main.go, when you catch the error (which you're already doing), bubble the error to the CLI with `log.Printf`.

Reply:
Refactored the whole function into main.go. Seems more appropriate there.